examples/ola-recorder.cpp

```diff
@@ -115,36 +121,62 @@ int VerifyShow(const string &filename) {
     return ola::EXIT_NOINPUT;
 
   map<unsigned int, unsigned int> frames_by_universe;
-  uint64_t total_time = 0;
 
-  unsigned int universe;
-  ola::DmxBuffer buffer;
-  unsigned int timeout;
+  ShowEntry entry;
   ShowLoader::State state;
+  ShowPlayer::PlaybackTime playback_pos = 0;
+  bool playing = false;
   while (true) {
-    state = loader.NextFrame(&universe, &buffer);
+    state = loader.NextEntry(&entry);
     if (state != ShowLoader::OK)
       break;
-    frames_by_universe[universe]++;
-
-    state = loader.NextTimeout(&timeout);
-    if (state != ShowLoader::OK)
+    playback_pos += entry.next_wait;
+    if (FLAGS_stop > 0 && playback_pos >= FLAGS_stop) {
+      // Compensate for overshooting the stop time
+      playback_pos -= (playback_pos - FLAGS_stop);
       break;
-    total_time += timeout;
+    }
+    if (!playing && playback_pos > FLAGS_start) {
+      // Found the start point
+      playing = true;
+    }
+    if (playing) {
+      frames_by_universe[entry.universe]++;
```

Comment:
This won't match the frames output if it's pre-cached frames? Perhaps increment during seek and then reset to 1 if non-zero when you currently set playing=true?

Reply:
That's true - now that multiple universes are handled properly, this needs some fixing.

I originally wanted to implement this by running the ShowPlayer in a sort of simulate mode to avoid these issues, but it's too closely coupled to having olad running for that to be workable...

Reply:
We do PIMPL ( https://en.cppreference.com/w/cpp/language/pimpl ) in a few other places. I think the only olad interfaces that needs are connect and send data and register a timeout, it ought to be possible to mock those up, but it might not be worth it.